### ISSUE TITLE
neo4j: brew upgrade removes database

### ISSUE THREAD

**Issue by tholu** [Homebrew/homebrew-core#37874]
- [x] are reporting a bug others will be able to reproduce and not asking a question or requesting software. If you're not sure or want to ask a question do so on our Discourse: https://discourse.brew.sh. To get software added or changed in Homebrew please file a [Pull Request](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)
- [x] have a problem with `brew install` (or `upgrade`, `reinstall`) a single, Homebrew/homebrew-core formula (not cask) on macOS? If it's a general `brew` problem please file this issue at Homebrew/brew: https://github.com/Homebrew/brew/issues/new/choose. If it's a Linux problem please file this issue at https://github.com/Linuxbrew/homebrew-core/issues/new/choose. If it's a `brew cask` problem please file this issue at https://github.com/Homebrew/homebrew-cask/issues/new/choose. If it's a tap (e.g. Homebrew/homebrew-php) problem please file this issue at the tap.
- [x] ran `brew update` and can still reproduce the problem?
- [x] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [ ] ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?
- [x] if `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

```
➜  ~ brew config
HOMEBREW_VERSION: 2.0.4-6-gae0332a
ORIGIN: https://github.com/Homebrew/brew
HEAD: ae0332a0f6fcf4846ef032ba217fb00ef66ccc48
Last commit: 17 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: bf98e5e1cf04e82e812be20634db15259a85ae82
Core tap last commit: 80 minutes ago
HOMEBREW_PREFIX: /usr/local
HOMEBREW_DEV_CMD_RUN: 1
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.q6OOYtIY01/org.macosforge.xquartz:0
HOMEBREW_LOGS: /Users/thomaslutz/Library/Logs/Homebrew
HOMEBREW_NO_ANALYTICS_THIS_RUN: 1
CPU: octa-core 64-bit skylake
Homebrew Ruby: 2.3.7 => /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/bin/ruby
Clang: 10.0 build 1000
Git: 2.17.2 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 7.54.0 => /usr/bin/curl
Java: 1.8.0_131, 1.8.0_112
macOS: 10.14.4-x86_64
CLT: 10.1.0.0.1.1539992718
Xcode: 10.1
XQuartz: 2.7.11 => /opt/X11
```

To help us debug your issue please explain:
- What you were trying to do (and why)

I upgraded neo4j from 3.5.0 to 3.5.3 with `brew upgrade`. 

- What happened (include command output)

```
➜  ~ brew upgrade
Updating Homebrew...
==> Auto-updated Homebrew!
Updated 1 tap (homebrew/core).
==> Upgrading 1 outdated packages:
neo4j 3.5.0 -> 3.5.3
==> Upgrading neo4j
==> Downloading https://neo4j.com/artifact.php?name=neo4j-community-3.5.3-unix.tar.gz
######################################################################## 100.0%
==> Caveats
To have launchd start neo4j now and restart at login:
  brew services start neo4j
Or, if you don't want/need a background service you can just run:
  neo4j start
==> Summary
🍺  /usr/local/Cellar/neo4j/3.5.3: 132 files, 109.0MB, built in 46 seconds
Removing: /usr/local/Cellar/neo4j/3.5.0... (182 files, 2.2GB)
Removing: /Users/thomaslutz/Library/Caches/Homebrew/neo4j--3.5.0.tar.gz... (97MB)
```

As the neo4j database is located in `/usr/local/Cellar/neo4j/3.5.0/libexec/data/`, it was removed without warning.

- What you expected to happen

I expected to keep my data and log files of version 3.5.0 for neo4j.

- Step-by-step reproduction instructions (by running `brew install` commands)

Install neo4j. Import some data. Wait for new version.

`brew upgrade`

Your neo4j database for the old version is gone.

**Comment by tholu** [comment]
Am i the only one who thinks this is an issue? Any pointers how to fix this, then I'll prepare a PR myself.

**Comment by SMillerDev** [comment]
You somehow have to point neo4j to a directory that isn't versioned. Maybe something like `/var/neo4j`? https://docs.brew.sh has some info, mainly in the cookbook, on how to point to that.

**Comment by tholu** [comment]
@SMillerDev Thanks for your input! As to your suggestion, I agree to put config files in `/usr/local/etc/neo4j/` and the data files into `/usr/local/var/neo4j`, this seems consistent with formulas like `mongodb` as well.

**Comment by tholu** [comment]
I configured the storage for log and data files in a similar way that `mongodb` does in #38657. I left out the config files for now.

**Comment by zbeekman** [comment]
@tholu is this resolved now that #38657 is merged?

**Comment by tholu** [comment]
Yes, should be!
